app/controllers/home_controller.rb

```diff
@@ -2,8 +2,20 @@ class HomeController < ApplicationController
 
 
   def index
+
+    @popular_tags = ActsAsTaggableOn::Tag.most_used(6)
+    if params[:tag].present?
+      @posts = Post.tagged_with(params[:tag]).paginate(:page => params[:page], :per_page => 5)
```

Comment:
please use JSON-style hash everywhere is possible (when key is symbol and doesn't require quoting)